pygments/lexers/arturo.py

```diff
@@ -0,0 +1,250 @@
+            (r'\.\w+', Name.Attribute),                         # attributes
+        ],
+
+        'inside-interpol': [
+            (r'\|', String.Interpol, '#pop'),
+            (r'[^|]+', using(this)),
+        ],
+        'inside-template': [
+            (r'\|\|\>', String.Interpol, '#pop'),
+            (r'[^|]+', using(this)),
+        ],
+        'string-escape': [
+            (words(('\\\\', '\\n', '\\t','\\"')), String.Escape)
+        ],
+
+        'inside-simple-string': [
+            include('string-escape'),
+            (r'\|', String.Interpol, 'inside-interpol'),        # Interpolation
+            (r'\<\|\|', String.Interpol, 'inside-template'),    # Templates
+            (r'"', String.Double, '#pop'),  # Closing Quote
+            (r'[^|"]+', String)             # String Content
+        ],
+        'inside-smart-string': [
+            include('string-escape'),
+            (r'\|', String.Interpol, 'inside-interpol'),        # Interpolation
+            (r'\<\|\|', String.Interpol, 'inside-template'),    # Templates
+            (r'\n', String.Single, '#pop'), # Closing Quote
+            (r'[^|\n]+', String)            # String Content
+        ],
+        'inside-safe-string': [
+            include('string-escape'),
+            (r'\|', String.Interpol, 'inside-interpol'),        # Interpolation
+            (r'\<\|\|', String.Interpol, 'inside-template'),    # Templates
+            (r'»»»', String.Double, '#pop'),    # Closing Quote
+            (r'[^|»]+', String)                 # String Content
+        ],
+        'inside-regex-string': [
+            (r'\\[sSwWdDbBZApPxucItnvfr0]+', String.Escape),
+            (r'\|', String.Interpol, 'inside-interpol'),        # Interpolation
+            (r'\<\|\|', String.Interpol, 'inside-template'),    # Templates
+            (r'\/\}', String.Single, '#pop'),   # Closing Quote
+            (r'[^|\/]+', String.Regex) ,        # String Content
+        ],
+        'inside-curly-verb-string': [
+            include('string-escape'),
+            (r'\|', String.Interpol, 'inside-interpol'),        # Interpolation
+            (r'\<\|\|', String.Interpol, 'inside-template'),    # Templates
+            (r'\:\}', String.Double, '#pop'),   # Closing Quote
+            (r'[\s\S]', String)                 # String Content
+        ],
+        'inside-curly-string': [
+            include('string-escape'),
+            (r'\|', String.Interpol, 'inside-interpol'),        # Interpolation
+            (r'\<\|\|', String.Interpol, 'inside-template'),    # Templates
+            (r'\}', String.Single, '#pop'), # Closing Quote
+            (r'[\s\S]', String),            # String Content
+        ],
+        'inside-eof-string': [
+            include('string-escape'),
+            (r'\|', String.Interpol, 'inside-interpol'),        # Interpolation
+            (r'\<\|\|', String.Interpol, 'inside-template'),    # Templates
+            (r'\Z$', String.Single, '#pop'),    # Closing Quote
```

Comment:
Isn't `\Z` end-of-string?

Reply:
- [x] [remove redundant end-of-string](https://github.com/pygments/pygments/pull/2259/commits/e60c6c873b7aec67125389efe1b700374444beca)